qiskit_experiments/framework/composite/composite_experiment.py

```diff
@@ -41,9 +42,10 @@ def __init__(
         """
         self._experiments = experiments
         self._num_experiments = len(experiments)
+        analysis = CompositeAnalysis([exp.analysis for exp in self._experiments])
```

Comment:
I agree with @yaelbh 's concern. Probably this will fix the problem.
```suggestion
        analyses = []
        for exp in self._experiments:
            analyses.append(exp.analysis)
            exp.analysis = None
```

Reply:
After that you won't be able anymore to run analysis on a sub-experiment, independently from it being part of a composite experiment.